react-basic-practice/src/pages/dashboard.tsx

```diff
@@ -1,10 +1,38 @@
-import React from "react";
-import { Box, useDisclosure } from "@chakra-ui/react";
-import { ICustomer } from "@types";
-import { Search, Button, Modal, Table, TableColumn } from "@components";
+import React, { useState } from "react";
+import { Box } from "@chakra-ui/react";
+import { ICustomer, ActionType } from "@types";
+import { Search, Button, Modal, Table, TableColumn, Form } from "@components";
 
 const Dashboard: React.FC = () => {
-  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [isOpen, setIsOpen] = useState(false);
+  const [modalTitle, setModalTitle] = useState("Add Customer");
+  const [viewData, setViewData] = useState<ICustomer | undefined>(undefined);
+
+  const handleOpenModal = (title: string, data?: ICustomer) => {
+    setModalTitle(title);
+    setViewData(data);
+    setIsOpen(true);
+  };
+
+  const handleCloseModal = () => {
+    setIsOpen(false);
+    setViewData(undefined);
+  };
+
+  const handleAddCustomer = () => {
+    handleOpenModal("Add Customer");
+  };
+
+  const handleAction = (type: ActionType, id?: number) => {
+    const customer = id && data.find((item) => item.id === id);
+
+    customer &&
+      handleOpenModal(
+        type === "Edit" ? "Edit Customer" : "View Customer",
```

Comment:
create a constant for Add/Edit/View title. Like:
```
const string FORM_TITLE = {
  VIEW: "View Customer",
...
}
```

Reply:
i done